cmd/launcher/launcher.go

```diff
@@ -170,13 +170,33 @@ func runLauncher(ctx context.Context, cancel func(), opts *launcher.Options) err
 		}
 	}
 
+	configStore, err := agentbbolt.NewStore(logger, db, osquery.ConfigBucket)
+	if err != nil {
+		return fmt.Errorf("failed to create '%s' KVStore: %w", osquery.ConfigBucket, err)
+	}
+
+	serverDataBucketConsumer, err := agentbbolt.NewStore(logger, db, osquery.ServerProvidedDataBucket)
+	if err != nil {
+		return fmt.Errorf("failed to create '%s' KVStore: %w", osquery.ServerProvidedDataBucket, err)
+	}
+
+	desktopFlagsBucketConsumer, err := agentbbolt.NewStore(logger, db, agentFlagsBucketName)
+	if err != nil {
+		return fmt.Errorf("failed to create '%s' KVStore: %w", agentFlagsBucketName, err)
+	}
+
+	osqueryHistoryStore, err := agentbbolt.NewStore(logger, db, osqueryInstanceHistory.OsqueryHistoryInstanceKey)
+	if err != nil {
+		return fmt.Errorf("failed to create '%s' KVStore: %w", osqueryInstanceHistory.OsqueryHistoryInstanceKey, err)
+	}
+
 	// init osquery instance history
-	if err := osqueryInstanceHistory.InitHistory(db); err != nil {
+	if err := osqueryInstanceHistory.InitHistory(osqueryHistoryStore); err != nil {
 		return fmt.Errorf("error initializing osquery instance history: %w", err)
 	}
 
 	// create the osquery extension for launcher. This is where osquery itself is launched.
-	extension, runnerRestart, runnerShutdown, err := createExtensionRuntime(ctx, db, client, opts)
+	extension, runnerRestart, runnerShutdown, err := createExtensionRuntime(ctx, db, configStore, serverDataBucketConsumer, desktopFlagsBucketConsumer, client, opts)
```

Comment:
Any time we add a new bucket, we'll have to init it here and then thread it through `createExtensionRuntime` => `osqueryRunnerOptions` / `grpcRunnerOptions` => `commonRunnerOptions` => `LauncherTables`, which feels a decent amount of boilerplate/a lot of locations to touch in order to do an otherwise fairly small update. I get that the whole point is that we don't want to keep passing the bbolt db around everywhere -- would we want to consider another option to reduce boilerplate? E.g. maybe a very small interface for something like `GetOrCreateBucket(bucketName string) types.KVStore` that we can use for areas like `LauncherTables` that require access to a couple different buckets.

Reply:
I think we'll want something, yeah. This might be a good corner to test with. 

I've been assuming we'll evolve a layer above `store`. But I don't know what it'll look like

Reply:
I've introduced `Storage.GetStore()` to address this! This change has helped in two ways - reducing boilerplate and argument lists, and ensuring there is a single instance of each store.